zcash/secant-android-wallet
# [#209] Compose dependency update

This PR contains:
- Update of the Compose dependency
- Align of our integration tests with the changes:
  - Add _integration_ folder to a few screen tests suits, if missing. In this folder we can have now for example these three files: 
     - `UpdateActivityViewTest` - using `createAndroidComposeRule<UiTestingActivity>()`
     - `UpdateViewIntegrationTest` - using `createComposeRule()`
     - `UpdateViewModelTest`
  - Split integration tests, which require activity to separate classes, as above
  - Remove unnecessary custom activity for each integration app part and create one custom common activity
- Implement Downloadable Fonts from Google Fonts
- Update Third party licenses documentation

@geffenz As we now use downloadable fonts, which are downloaded in runtime on the first app run with the internet connection, there can be a minor case a user starts the app on the first time without the connection and then a default font is used. The Roboto font is default on most Android devices today. Here is how it ends up in our app - I believe it's good enough. The left one is our **Rubik** and the right one is the **Roboto**.

![default_font](https://user-images.githubusercontent.com/9017390/173401993-de2d12d0-9a95-4dc7-8d46-7706ebfc1dcd.png)

@ccjernigan The runtime-downloaded Rubik fonts seem to be the same as the ones from our res folder. Just FYI - only one visible change happened - our `Typography` definition contains `headlineLarge`, which used **W600** weight of Rubik font, which wasn't provided in our res folder, so I assume that the **Medium** weight was used. Now in the new implementation we also provide **SemiBold** (W600) weight, which changes our headlines like this:

![headlines](https://user-images.githubusercontent.com/9017390/173399483-a97d9968-aed2-4efc-ae97-291b2c65e749.png)

Closes https://github.com/zcash/secant-android-wallet/issues/209, https://github.com/zcash/secant-android-wallet/issues/466.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [ ] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules.
- [x] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [x] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._


# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [x] Automated tests: Did you review the automated tests?
- [x] Manual tests: Did you review the manual tests?
- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
- [x] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._
